### DISCUSSION TITLE
How do you add Sections to Sections?

### DISCUSSION THREAD

**Discussion by slussinator** [ctreffe/alfred#152] [Alfred Q&A]
I would like to create a subsection within a section. What is the easiest way to do this? 
Also: Is it possible to create sub-subsections?

**Comment by storyteller-dev** [comment]
Yes, both is possible and quite straightforward. You simply append a new section to an existing section instead of the experiment instance:

```Python
import alfred3 as al

exp = al.Experiment()

# Our first section is appended directly to the experiment
exp += al.Section(name="base")

# To create a subsection, append a new section to exp.base instead of exp
exp.base += al.Section(name="subsection_1")
exp.base.subsection_1 += al.Page(name="page_1")
exp.base.subsection_1.page_1 += al.Text("Hello World")

# Do it again to have a second subsection
exp.base += al.Section(name="subsection_2")
exp.base.subsection_2 += al.Page(name="page_2")
exp.base.subsection_2.page_2 += al.Text("Hello World again")

# And to add a subsubsection to a subsection, append a new section to the subsection
exp.base.subsection_2 += al.Section(name="subsubsection")
exp.base.subsection_2.subsubsection += al.Page(name="page_3")
exp.base.subsection_2.subsubsection.page_3 += al.Text("Hello World for the third time")
```

**Reply by slussinator** [reply]
Great, thanks! This is how I add sections and subsections in instance style. Could you explain how to do it in class style as well?

**Reply by ctreffe** [reply]
Good point, let's extend @mate-code's answer. The easiest solution to add sections and pages to other sections is to use the _member_ decorator with the parameter _of\_section_ ([see the docs for additional information](https://alfredo3.psych.bio.uni-goettingen.de/docs/generated/alfred3.experiment.Experiment.member.html?highlight=decorator)):

```python
import alfred3 as al

exp = al.Experiment()

# Our first section is appended directly to the experiment
@exp.member
class Main(al.Section):
    ...

# Now we can add a subsection to the first section
@exp.member(of_section="Main")
class Sub(al.Section):
    ...

# Adding pages to a section works the same way
@exp.member(of_section="Sub")
class Page_1(al.Page):
    title = "My First Page"

```

This way of using the _of\_section_ parameter is also described with another code example in [_How to use pages_](https://alfredo3.psych.bio.uni-goettingen.de/docs/howto_pages.html?highlight=page%20name).
